### PR TITLE
TMDM-13896  [Studio] Update API to support "Category" annotation

### PR DESCRIPTION
**Pull request by hbhong**
**What is the current behavior?** (You should also link to an open issue here)
Current API does not support for editing "Category"

link: https://jira.talendforge.org/browse/TMDM-13896

**What is the new behavior?**
Provide support for "Category" annotation in the studio side

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/utils/XSDAnnotationsStructure.java [hunk 1]
```diff
@@ -816,6 +823,239 @@ public void removeWorkflow(String pattern) {
...
+        for (String field : fieldMap.keySet()) {
+            String category = fieldMap.get(field);
+            if (!categoryFieldsMap.containsKey(category)) {
+                categoryFieldsMap.put(category, new ArrayList<String>());
+            }
+            categoryFieldsMap.get(category).add(field);
+        }
+
+        for (Element element : annotation.getApplicationInformation()) {
+
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/utils/XSDAnnotationsStructure.java]: Remove this blank line

File: test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import org.eclipse.xsd.XSDSchema;
+import org.junit.Test;
+import org.w3c.dom.Element;
+
+import com.amalto.workbench.models.IAnnotationConst;
+
+@SuppressWarnings("nls")
+public class CategoryTest {
+
+    private static Logger log = Logger.getLogger(CategoryTest.class);
```
Comment (lpw82gf) [inline test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java]: log -> LOG

File: test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
+package com.amalto.workbench.utils;
```
Comment (lpw82gf) [inline test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java]: Copyright?

Comment (hbhong) [reply test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java]: OK, I'll fix them

File: test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java (hunk 2 above)
Comment (lpw82gf) [reply test/plugins/org.talend.mdm.workbench.test/src/main/java/com/amalto/workbench/utils/CategoryTest.java]: `private static final Logger LOG`

**Pull request merged**